File: Domain/Activity.cs
using System.ComponentModel.DataAnnotations;

namespace Domain;

public class Activity
{
    [Required]
    public Guid Id { get; set; } 
    [Required]
    public string Title { get; set; }
    public DateTime Date { get; set; }
    public required string Description { get; set; }
    public required string Category { get; set; }
    public bool IsCancelled { get; set; }
    /// <summary>
    /// location of the activity
    /// </summary>
    public required string City { get; set; }
    public required string Venue { get; set; }
    public double Latitude { get; set; }
    public double Longitude { get; set; }

    // navigation properties
    public ICollection<ActivityAttendee> Attendees { get; set; } = [];
}

File: vite-project/src/app/models/Types/activity.ts
import { Profile } from "./profile";

export interface Activity {
  id: string;
  title: string;
  date: Date | null;
  description: string;
  category: string;
  city: string;
  venue: string;
  latitude: number | 0;
  longitude: number;
  attendees: Profile[];
  isGoing: boolean;
  isCancelled: boolean;
  isHost: boolean;
  hostUsername: string;
  hostId: string;
  hostDisplayName: string;
  hostImageUrl: string;
}
